### PR TITLE
6766: remove sphinx

### PR DESCRIPTION
**Pull request by meltheadorable**

### CODE REVIEW

File: app/models/answer.rb [hunk 1]
```diff
@@ -58,6 +60,19 @@ class Answer < ApplicationRecord
...
+    against: :value,
+    associated_against: {
+      option: :canonical_name,
+      options: :canonical_name,
+    },
+    using: {
+      tsearch: {
+        prefix: true
+      }
+    }
```
Comment (hooverlunch) [inline app/models/answer.rb]: oooh!

File: docs/development-setup.md [hunk 2]
```diff
@@ -17,12 +17,6 @@ Note to install the software below we recommend the following package managers:
 1. **PostgreSQL 9.4+**
   - Create an empty database for use by the app (typically *elmo_development*)
 
-1. **Sphinx 2.0.6+**
-  - Sphinx is an open source search engine.
-  - It should be available through any of the package managers listed above. If not it can be built from source.
-  - The Rails Gem that talks to Sphinx is called Thinking Sphinx.
-  - The [Thinking Sphinx site](http://pat.github.io/thinking-sphinx/) is a good place to go for troubleshooting and documentation.
-
```
Comment (hooverlunch) [inline docs/development-setup.md]: Ahhhhhhhhh

File: docs/production-setup.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ For security reasons, it is not recommended to install ELMO as the `root` user.
 ### Install dependencies
 
     sudo apt-get update && sudo apt-get -y upgrade
-    sudo apt-get -y install nano git-core curl zlib1g-dev build-essential libssl-dev libreadline-dev libyaml-dev libsqlite3-dev sqlite3 libxml2-dev libxslt1-dev libcurl4-openssl-dev libffi-dev libmysqlclient-dev python-software-properties nodejs sphinxsearch memcached imagemagick
```
Comment (hooverlunch) [inline docs/production-setup.md]: Good catch to remove this!

File: spec/features/responses_index_spec.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@
   let(:form) { create(:form, :published, name: "TheForm") }
   let(:response_link) { Response.first.decorate.shortcode }
 
-  scenario "returning to index after response loaded via ajax", js: true, sphinx: true do
```
Comment (hooverlunch) [inline spec/features/responses_index_spec.rb]: Hooray for getting rid of all these!

**Comment by hooverlunch** [comment]
Update: Seems we may need an index: https://stackoverflow.com/questions/19310294/search-too-slow-how-do-i-reindex-postgres-database-in-a-rails-app-using-pg-sear

**Comment by hooverlunch** [comment]
Update: I went ahead and fixed the search examples and hints. Let me know if they look right to you.

**Comment by meltheadorable** [comment]
Looks like we may need to do this https://github.com/Casecommons/pg_search/wiki/Building-indexes

**Comment by hooverlunch** [comment]
Holy crap, this is the query that pg_search is building:

```
SELECT "answers"."id" FROM "answers"
  INNER JOIN "responses" ON "responses"."id" = "answers"."response_id" AND "responses"."deleted_at" IS NULL
  INNER JOIN "form_items" ON "form_items"."id" = "answers"."questioning_id" AND "form_items"."type" IN ('Questioning') AND "form_items"."deleted_at" IS NULL
  INNER JOIN (
    SELECT "answers"."id" AS pg_search_id, 
      (ts_rank((to_tsvector('simple', coalesce("answers"."value"::text, '')) || to_tsvector('simple', coalesce(pg_search_45f94cae7cae1c18e7940d.pg_search_152ae1a1f9a971ba013dc4::text, '')) || to_tsvector('simple', coalesce(pg_search_608849bc87d1779b75f2c7.pg_search_03982092fd8483b3c5e5ea::text, ''))), (to_tsquery('simple', ''' ' || 'ballots' || ' ''' || ':*')), 0)) AS rank 
    FROM "answers" 
    LEFT OUTER JOIN (
      SELECT "answers"."id" AS id, "options"."canonical_name"::text AS pg_search_152ae1a1f9a971ba013dc4 FROM "answers"
        INNER JOIN "options" ON "options"."id" = "answers"."option_id" AND "options"."deleted_at" IS NULL
    ) pg_search_45f94cae7cae1c18e7940d ON pg_search_45f94cae7cae1c18e7940d.id = "answers"."id" 
    LEFT OUTER JOIN (
      SELECT "answers"."id" AS id, string_agg("options"."canonical_name"::text, ' ') AS pg_search_03982092fd8483b3c5e5ea FROM "answers"
        INNER JOIN "choices" ON "choices"."answer_id" = "answers"."id" AND "choices"."deleted_at" IS NULL
        INNER JOIN "options" ON "options"."id" = "choices"."option_id" AND "options"."deleted_at" IS NULL GROUP BY "answers"."id"
    ) pg_search_608849bc87d1779b75f2c7 ON pg_search_608849bc87d1779b75f2c7.id = "answers"."id"
    WHERE (((
      to_tsvector('simple', coalesce("answers"."value"::text, '')) || 
      to_tsvector('simple', coalesce(pg_search_45f94cae7cae1c18e7940d.pg_search_152ae1a1f9a971ba013dc4::text, '')) || 
      to_tsvector('simple', coalesce(pg_search_608849bc87d1779b75f2c7.pg_search_03982092fd8483b3c5e5ea::text, ''))) @@ (
        to_tsquery('simple', ''' ' || 'ballots' || ' ''' || ':*')
      )
    ))
  ) AS pg_search_677fe21b65f4a6cf1d1058 ON "answers"."id" = pg_search_677fe21b65f4a6cf1d1058.pg_search_id 
  WHERE "answers"."deleted_at" IS NULL AND "responses"."mission_id" = $1  
  ORDER BY pg_search_677fe21b65f4a6cf1d1058.rank DESC, "answers"."id" ASC
```

**Pull request merged**